lib/spack/spack/build_systems/README-intel.rst

```diff
@@ -0,0 +1,650 @@
+       simplifies the process.  In addition, as an administrator it is good
+       practice to know WHERE your licenses are saved on your system.
+       License files are plain text files with a .lic extension.  By default
+       these are saved in /opt/intel/licenses which is searched by default.
+       If you save your license elsewhere, perhaps under an NFS folder, set
+       environment variable **INTEL_LICENSE_FILE** to the full path to your
+       license file prior to starting the installation or use the
+       configuration file directive ``ACTIVATION_LICENSE_FILE`` to specify the
+       full pathname to the license file.
+    
+       Options for ``ACTIVATION`` are ``{ exist_lic, license_file, server_lic,
+       serial_number, trial_lic }``
+    
+    ``exist_lic``
+       directs the installer to search for a valid license on the server.
+       Searches will utilize the environment variable **INTEL_LICENSE_FILE**,
+       search the default license directory /opt/intel/licenses, or use the
+       ``ACTIVATION_LICENSE_FILE`` directive to find a valid license file.
+    
+    ``license_file``
+       is similar to exist_lic but directs the installer to use
+       ``ACTIVATION_LICENSE_FILE`` to find the license file.
+    
+    ``server_lic``
+       is similar to exist_lic and exist_lic but directs the installer that
+       this is a client installation and a floating license server will be
+       contacted to active the product.  This option will contact your
+       floating license server on your network to retrieve the license
+       information.  BEFORE using this option make sure your client is
+       correctly set up for your network including all networking, routing,
+       name service, and firewall configuration.  Insure that your client has
+       direct access to your floating license server and that firewalls are
+       set up to allow TCP/IP access for the 2 license server ports.
+       server_lic will use **INTEL_LICENSE_FILE** containing a port@host format
+       OR a client license file.  The formats for these are described here
+       https://software.intel.com/en-us/articles/licensing-setting-up-the-client-floating-license
+    
+    ``serial_number``
+       directs the installer to use directive ``ACTIVATION_SERIAL_NUMBER`` for
+       activation.  This method will require the installer to contact an
+       external Intel activation server over the Internet to confirm your
+       serial number.  Due to user and company firewalls, this method is more
+       complex and hence error prone of the available activation methods.  We
+       highly recommend using a license file or license server for activation
+       instead.
+    
+    ``trial_lic``
+       is used only if you do not have an existing license and intend to
+       temporarily evaluate the compiler.  This method creates a temporary
+       trial license in Trusted Storage on your system.
+    
+    ...
```

Comment:
I think a brief summary would be really helpful, e.g. which one of these is `spack` going to execute?  For the license file I am about to try one of these, it would be nice to do a "definition list" summary here.

```rst
If you are having ``spack`` install the intel compiler suite, ``spack`` will download the installer for you.  However, **before** executing ``spack install intel-<pkg>``, you will need to make sure the Intel installer(s) can find your license.

**License Server**
    ... what should users do to coordinate the license server with ``spack`` for installing the compiler suite? ...

**License File**
    If you are using a single-user license you can obtain a license file from Intel by logging in and `following Intel's instructions <https://software.intel.com/en-us/articles/resend-license-file>`_.  With ``downloaded_name.lic`` downloaded on your local machine, you may either

    1. Create the directory ``/opt/intel/licenses`` and copy ``downloaded_name.lic`` there (the Intel installer will search here by default).
    2. Set the environment variable ``export INTEL_LICENSE_FILE="/path/to/downloaded_name.lic"``.
```

Reply:
@svenevs: Looks good - I'll integrate this into user instructions to live at lib/spack/docs/build_systems/intelpackage.rst , per #5015 pinged by @adamjstewart .

Reply:
Ok so these instructions did not work.  When I performed `spack install intel-parallel-studio@cluster.2017.0 +advisor+inspector+vtune`, `spack` will open up `$EDITOR` and ask you to put in the license information.  I had `INTEL_LICENSE_FILE` exported but that still failed, and if I just put the line `/opt/intel/licenses/mylicense.lic` in the prompt that will also fail.  I had to paste my license into the prompt `spack` brings up.

Now that the package is downloaded I can redo this a bunch of times with relative ease.  What is the expected workflow here / how did you do it with the license server?  As in when you got to this screen, what did you enter?

```
# A license is required to use intel-parallel-studio.
#
# The recommended solution is to store your license key in this global
# license file. After installation, the following symlink(s) will be
# added to point to this file (relative to the installation prefix):
#
#       Licenses/license.lic
#       /opt/spack_intel/opt/spack/linux-fedora25-x86_64/gcc-6.4.1/intel-parallel-studio-cluster.2017.0-xrbcvside3freayu7kztskrm7476ye45/compilers_and_librari
es/linux/bin/intel64/license.lic
#       inspector/licenses/license.lic
#       advisor/licenses/license.lic
#       vtune_amplifier_xe/licenses/license.lic
#
# Alternatively, use one of the following environment variable(s):
#
#       INTEL_LICENSE_FILE
#
# If you choose to store your license in a non-standard location, you may
# set one of these variable(s) to the full pathname to the license file, or
# port@host if you store your license keys on a dedicated license server.
# You will likely want to set this variable in a module file so that it
# gets loaded every time someone tries to use intel-parallel-studio.
#
# For further information on how to acquire a license, please refer to:
#
#       https://software.intel.com/en-us/articles/intel-license-manager-faq
#
# You may enter your license below.
>>> what goes here? <<<
```

It seems that `spack` is explicitly forcing in `$SPACK_ROOT/etc/spack/licenses/license.lic` (see all the symlinks being created):

```console
$ spack install intel-parallel-studio@cluster.2017.0 +advisor+inspector+vtune                                                            
==> Installing intel-parallel-studio
==> Using cached archive: /opt/spack_intel/var/spack/cache/intel-parallel-studio/intel-parallel-studio-cluster.2017.0.tgz
==> Staging archive: /opt/spack_intel/var/spack/stage/intel-parallel-studio-cluster.2017.0-xrbcvside3freayu7kztskrm7476ye45/parallel_studio_xe_2017.tgz
==> Created stage in /opt/spack_intel/var/spack/stage/intel-parallel-studio-cluster.2017.0-xrbcvside3freayu7kztskrm7476ye45
==> No patches needed for intel-parallel-studio
==> Building intel-parallel-studio [IntelPackage]
==> Added global license file /opt/spack_intel/etc/spack/licenses/intel/license.lic
==> Executing phase: 'configure'
==> Executing phase: 'install'
==> Added local symlink /opt/spack_intel/opt/spack/linux-fedora25-x86_64/gcc-6.4.1/intel-parallel-studio-cluster.2017.0-xrbcvside3freayu7kztskrm7476ye45/Licenses/license.lic to global license file
==> Added local symlink /opt/spack_intel/opt/spack/linux-fedora25-x86_64/gcc-6.4.1/intel-parallel-studio-cluster.2017.0-xrbcvside3freayu7kztskrm7476ye45/compilers_and_libraries_2017.0.098/linux/bin/intel64/license.lic to global license file
==> Added local symlink /opt/spack_intel/opt/spack/linux-fedora25-x86_64/gcc-6.4.1/intel-parallel-studio-cluster.2017.0-xrbcvside3freayu7kztskrm7476ye45/inspector_2017.1.0.475470/licenses/license.lic to global license file
==> Added local symlink /opt/spack_intel/opt/spack/linux-fedora25-x86_64/gcc-6.4.1/intel-parallel-studio-cluster.2017.0-xrbcvside3freayu7kztskrm7476ye45/advisor_2017.1.0.477503/licenses/license.lic to global license file
==> Added local symlink /opt/spack_intel/opt/spack/linux-fedora25-x86_64/gcc-6.4.1/intel-parallel-studio-cluster.2017.0-xrbcvside3freayu7kztskrm7476ye45/vtune_amplifier_xe_2017.0.2.478468/licenses/license.lic to global license file
==> Successfully installed intel-parallel-studio
  Fetch: 6.95s.  Build: 3m 42.31s.  Total: 3m 49.27s.
[+] /opt/spack_intel/opt/spack/linux-fedora25-x86_64/gcc-6.4.1/intel-parallel-studio-cluster.2017.0-xrbcvside3freayu7kztskrm7476ye45
```

Reply:
@svenevs: Thanks for testing. 

What does `grep ACTIVATION /opt/spack_intel/...<HASH>/.spack/silent.cfg`  give?

The native installer looks into `/opt/intel/licenses/` by default, so if you can write there, ideally you should not need to have to edit anything in Spack,
and neither `$SPACK_ROOT/etc/spack/licenses/license.lic` nor its symlinks should be needed, though the mere placeholder content will not hurt.

I should be able to add logic to preempt the prompt, as much as I'd like to avoid assumptions and duplications of logic that the installer does.

Can you share a disabled version of your `/opt/intel/licenses/mylicense.lic` file, e.g., as run through `sed 's/[0-9A-F]\{4,\}/____/g' `?